From f60df0913078a943eb5120ef65b8e0430b922516 Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Thu, 19 Nov 2015 19:49:11 +0000
Subject: [PATCH 157/178] staging: etnaviv: remove unused
 etnaviv_gem_get_iova()

etnaviv_gem_get_iova() is never used, the _locked version is always
used in preference to this one as we're always holding the struct_mutex
everywhere.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_drv.h |  2 --
 drivers/staging/etnaviv/etnaviv_gem.c | 23 -----------------------
 2 files changed, 25 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_drv.h b/drivers/staging/etnaviv/etnaviv_drv.h
index a296783..c100b2d 100644
--- a/drivers/staging/etnaviv/etnaviv_drv.h
+++ b/drivers/staging/etnaviv/etnaviv_drv.h
@@ -76,8 +76,6 @@ int etnaviv_gem_fault(struct vm_area_struct *vma, struct vm_fault *vmf);
 int etnaviv_gem_mmap_offset(struct drm_gem_object *obj, u64 *offset);
 int etnaviv_gem_get_iova_locked(struct etnaviv_gpu *gpu,
 	struct drm_gem_object *obj, u32 *iova);
-int etnaviv_gem_get_iova(struct etnaviv_gpu *gpu, struct drm_gem_object *obj,
-	u32 *iova);
 void etnaviv_gem_put_iova(struct drm_gem_object *obj);
 struct sg_table *etnaviv_gem_prime_get_sg_table(struct drm_gem_object *obj);
 void *etnaviv_gem_prime_vmap(struct drm_gem_object *obj);
diff --git a/drivers/staging/etnaviv/etnaviv_gem.c b/drivers/staging/etnaviv/etnaviv_gem.c
index bf0295b..0478967 100644
--- a/drivers/staging/etnaviv/etnaviv_gem.c
+++ b/drivers/staging/etnaviv/etnaviv_gem.c
@@ -273,29 +273,6 @@ int etnaviv_gem_get_iova_locked(struct etnaviv_gpu *gpu,
 	return ret;
 }
 
-int etnaviv_gem_get_iova(struct etnaviv_gpu *gpu, struct drm_gem_object *obj,
-	u32 *iova)
-{
-	struct etnaviv_gem_object *etnaviv_obj = to_etnaviv_bo(obj);
-	struct etnaviv_vram_mapping *mapping =
-			etnaviv_gem_get_vram_mapping(etnaviv_obj, gpu->mmu);
-	int ret;
-
-	/* this is safe right now because we don't unmap until the
-	 * bo is deleted:
-	 */
-	if (mapping) {
-		*iova = mapping->iova;
-		return 0;
-	}
-
-	mutex_lock(&obj->dev->struct_mutex);
-	ret = etnaviv_gem_get_iova_locked(gpu, obj, iova);
-	mutex_unlock(&obj->dev->struct_mutex);
-
-	return ret;
-}
-
 void etnaviv_gem_put_iova(struct drm_gem_object *obj)
 {
 	/*
-- 
2.6.2

